conftest.py

```diff
@@ -4,11 +4,38 @@
 import requests
 from typing import Callable
 import logging
+from dataclasses import dataclass
 
 
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int  # RuntimeError: Sensor didn't reset to factory property if float
+
+
+    def __post_init__(self):
+        if not isinstance(self.name, str) or self.name == "":
```

Comment:
It's good that you're looking for ways to shorten the code. In this case, however, by doing an `or` you removed the distinction between a `TypeError` and a `ValueError`. Generally this is not something we want to do. Same goes for other validations.

Reply:
checks separeted